src/main/webapp/resources/js/pages/projects/samples-metadata-import/redux/importReducer.ts

```diff
@@ -91,79 +95,173 @@ export const saveMetadata = createAsyncThunk<
           label: header.name,
           restriction: header.targetRestriction,
         })),
+    }).catch((error) => {
+      throw new Error(error.response.data.error);
     });
 
-    const chunkSize = 100;
-    for (let i = 0; i < metadata.length; i = i + chunkSize) {
-      const promises: Promise<void>[] = [];
-      for (let j = i; j < i + chunkSize && j < metadata.length; j++) {
-        const metadataItem: MetadataItem = metadata[j];
-        const index: string = metadataItem.rowKey;
-        if (
-          selectedMetadataKeys.includes(index) &&
-          metadataSaveDetails[index]?.saved !== true
-        ) {
-          const name: string = metadataItem[sampleNameColumn];
-          const metadataFields: FieldUpdate[] = Object.entries(metadataItem)
-            .filter(
-              ([key]) =>
-                headers.map((header) => header.name).includes(key) &&
-                key !== sampleNameColumn
-            )
-            .map(([key, value]) => ({
-              field: key,
-              value,
-              restriction: headers.filter((header) => header.name === key)[0]
-                .targetRestriction,
-            }));
-          const sampleId = metadataValidateDetails[index].foundSampleId;
-          if (sampleId) {
-            promises.push(
-              updateSample({
-                projectId,
-                sampleId,
-                body: {
-                  name,
-                  metadata: metadataFields,
-                },
-              })
-                .then(() => {
-                  metadataSaveDetails[index] = { saved: true };
-                })
-                .catch((error) => {
-                  metadataSaveDetails[index] = {
-                    saved: false,
-                    error: error.response.data.error,
-                  };
-                })
-            );
-          } else {
-            promises.push(
-              createSample({
-                projectId,
-                body: {
-                  name,
-                  metadata: metadataFields,
-                },
-              })
-                .then(() => {
-                  metadataSaveDetails[index] = { saved: true };
-                })
-                .catch((error) => {
-                  metadataSaveDetails[index] = {
-                    saved: false,
-                    error: error.response.data.error,
-                  };
-                })
-            );
-          }
-        }
-      }
-      await Promise.all(promises).then(() => {
-        dispatch(
-          setMetadataSaveDetails(Object.assign({}, metadataSaveDetails))
+    const selectedSampleList = metadata.filter((metadataItem) => {
+      const name: string = metadataItem[sampleNameColumn];
+      return (
+        selectedMetadataKeys.includes(metadataItem.rowKey) &&
+        metadataSaveDetails[name]?.saved !== true
```

Comment:
This is always going to evaluate the same, I.e. the sample name will never exist in the `metadataSaveDetails` object, since you are creating it on line 88.

Reply:
My mistake. Please see 4712e69ac8787e4238a85a17bec852d98e153c06.